### DISCUSSION TITLE
Using `connection_limit` in combination with `pg_bouncer=true`

### DISCUSSION THREAD

**Discussion by FarazPatankar** [prisma/prisma#19034] [Q&A] [answered]
Hello hello,

I've been reading through the docs for [increasing the pool size](https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#increasing-the-pool-size) and [configuring the client with PgBouncer](https://www.prisma.io/docs/guides/performance-and-optimization/connection-management/configure-pg-bouncer) but I can't seem to find anything on how to use them in combination with each other. Is that even possible? And if it is, is it as simple as using both the query params?

For context, we run a temporal worker that calls the DB and has been running to the `Timed out fetching a new connection from the connection pool.` error and wanted to increase the `connection_limit` and see if that fixes the issue.

Also, what's the best way to test if the change to the connection limit has been reflected? Is there a function or a value within the client which returns what the `connection_limit` is?

Edit: I noticed [this discussion](https://github.com/prisma/prisma/discussions/18290) and the `metrics` feature but looks like it's only available from `3.15.0` and we're still on `2.23.0`. Is there anything similar available on an older version?

**Comment by ludralph** [comment, marked as answer]
Hi @FarazPatankar 

> I've been reading through the docs for [increasing the pool size](https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#increasing-the-pool-size) and [configuring the client with PgBouncer](https://www.prisma.io/docs/guides/performance-and-optimization/connection-management/configure-pg-bouncer) but I can't seem to find anything on how to use them in combination with each other. Is that even possible? And if it is, is it as simple as using both the query params?

You can combine both the connection_limit and pg-bouncer like this 
`postgresql://johndoe:mypassword@localhost:5432/mydb?schema=public&connection_limit=5&pgbouncer=true`. It is as simple as using both the query params.

>Also, what's the best way to test if the change to the connection limit has been reflected? Is there a function or a value within the client which returns what the connection_limit is?
Edit: I noticed https://github.com/prisma/prisma/discussions/18290 and the metrics feature but looks like it's only available from 3.15.0 and we're still on 2.23.0. Is there anything similar available on an older version?

In older Prisma versions, you might have to write raw SQL to determine the number of connections. I'll advise you to update to more recent versions of Prisma.

**Reply by FarazPatankar** [reply]
Thanks for the response, @ludralph. Do you know the raw SQL I could write to determine Prisma-specific values for open connections, connection limits and max connections?

**Reply by ludralph** [reply]
hi @FarazPatankar, if you're using postgres, this query can be helpful

```sql
select count(*) from pg_stat_activity;
```
You can also take a look at this [Stackoverflow](https://stackoverflow.com/questions/5267715/right-query-to-get-the-current-number-of-connections-in-a-postgresql-db) post.

**Reply by ludralph** [reply]
Hi @FarazPatankar 

Do you have any more questions about this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by ludralph** [reply]
Hi @FarazPatankar 

Do you have any more questions about this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!
